### PR TITLE
Table & visualization view

### PR DESCRIPTION
**Pull request by gmrukwa**
* collapsing of `npm list` result during the build
* simple definition of the table-view
* simple definition of a factory-alike component

### CODE REVIEW

File: src/app/visualization-view/table-view/table-view.component.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  ngOnInit() {
+    if (this.columns !== undefined) {
+      this.columnNames = this.columns.map(columnDefinition => columnDefinition.key);
+    }
+  }
+
+  ngOnChanges() {
+    if (this.columns !== undefined) {
+      this.columnNames = this.columns.map(columnDefinition => columnDefinition.key);
+    }
```
Comment (Sebaxp95) [inline src/app/visualization-view/table-view/table-view.component.ts]: Code from line 37,38 and 43,44 should be in one function, or maybe setter from this https://ngdev.space/angular-2-input-property-changes-detection-3ccbf7e366d2 ?
I do not check this solution, so I do not know if it works in our case.

**Comment by gmrukwa** [comment]
[table-view](https://trello.com/c/u1kL84qf/424-table-view)

File: src/app/visualization-view/visualization-view.module.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+import {
+  MatTableModule,
+} from '@angular/material';
```
Comment (Sebaxp95) [inline src/app/visualization-view/visualization-view.module.ts]: Style of import is different

Comment (gmrukwa) [reply src/app/visualization-view/visualization-view.module.ts]: I've normalized it in my IDE.

**Pull request merged**